sql-plugin/src/main/java/com/nvidia/spark/rapids/UnsafeRowToColumnarBatchIterator.java

```diff
@@ -163,7 +166,16 @@ public ColumnarBatch next() {
     }
     try (NvtxRange ignored = buildRange;
          ColumnVector cv = devColumn;
-         Table tab = Table.convertFromRows(cv, rapidsTypes)) {
+         // We are branching over the output.length to know which kernel to call.
+         // If output.length < 100 we call the fixed-width optimized version, otherwise the
```

Comment:
It is really odd to have all of these comments explaining the limitations of the fixed width optimized here, when the check for it is in a entirely different place. Could we do one of the following

1. Get rid of `isOptimizedForFixedWidth` and instead use `rapidsTypes.length` to decide which to call. If this is a problem for testing, then we can have an test only constructor that exposes an override for the check.
2. move the comments to GpuRowToColumnarExec.scala where isOptimizedForFixedWidth is calculated.

I personally prefer option 1, but it is more code.

Reply:
I have made the requested changes to `UnsafeRowToColumnarBatchIterator`.  And a similar change was made to the `GpuColumnarToRowExec`. PTAL